src/evict/evict_lru.c

```diff
@@ -1720,10 +1720,22 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 			 * Try a different walk start point next time if a
 			 * walk gave up.
 			 */
-			btree->evict_walk_state =
```

Comment:
@keithbostic I preferred the original version of this code, since it meant we didn't need to update another section of code when adding a new walk state - it'd be easy to forget to update the new case statement, and it's not an obvious bug. How horrible are the required casts?